mlflow/tracking/client.py

```diff
@@ -1053,24 +1053,27 @@ def log_figure(self, run_id, figure, artifact_file):
             run = client.create_run(experiment_id="0")
             client.log_figure(run.info.run_id, fig, "figure.html")
         """
-        MATPLOTLIB_FIGURE_REGEXP = r"^matplotlib\..*Figure$"
-        PLOTLY_FIGURE_REGEXP = r"^plotly\..*Figure$"
 
-        qual_class_name = figure.__class__.__module__ + "." + figure.__class__.__name__
-        if not any(
-            re.search(regexp, qual_class_name)
-            for regexp in [MATPLOTLIB_FIGURE_REGEXP, PLOTLY_FIGURE_REGEXP]
-        ):
-            raise TypeError("Unsupported figure object type: '{}'".format(type(figure)))
+        def is_matplotlib_figure(fig):
```

Comment:
Can we make this and `is_plotly_figure` private methods? e.g. `_is_plotly_figure`?